modules/videoio/src/cap_v4l.cpp

```diff
@@ -589,8 +594,14 @@ static void v4l2_create_frame(CvCaptureCAM_V4L *capture) {
     }
 
     /* Set up Image data */
-    cvInitImageHeader(&capture->frame, size, IPL_DEPTH_8U, channels);
+    switch(capture->palette) {
+    case V4L2_PIX_FMT_Y16:
+        cvInitImageHeader(&capture->frame, size, IPL_DEPTH_16U, channels);
```

Comment:
Add
```
int depth = IPL_DEPTH_8U;
```
near
```
int channels = 3;
```
above and use it in `cvInitImageHeader`.

In case of `!capture->convert_rgb` and `capture->palette == V4L2_PIX_FMT_Y16` change `depth`, `channels` (to 1?) and size(?) before this line (take a look on switch before).